google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/GeneratedUpdateTableStatementsTests.java

```diff
@@ -0,0 +1,123 @@
+
+import com.google.cloud.spanner.hibernate.entities.Employee;
+import com.mockrunner.mock.jdbc.JDBCMockObjectFactory;
+import com.mockrunner.mock.jdbc.MockConnection;
+import java.sql.SQLException;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.boot.Metadata;
+import org.hibernate.boot.MetadataSources;
+import org.hibernate.boot.registry.StandardServiceRegistry;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.junit.Before;
+import org.junit.Test;
+
+public class GeneratedUpdateTableStatementsTests {
+
+  private StandardServiceRegistry registry;
+
+  private JDBCMockObjectFactory jdbcMockObjectFactory;
+
+  private MockConnection mockConnection;
+
+  /**
+   * Set up the metadata for Hibernate to generate schema statements.
+   */
+  @Before
+  public void setup() {
+    this.jdbcMockObjectFactory = new JDBCMockObjectFactory();
+    this.jdbcMockObjectFactory.registerMockDriver();
+
+    mockConnection = this.jdbcMockObjectFactory.createMockConnection();
+
+    this.jdbcMockObjectFactory
+        .getMockDriver()
+        .setupConnection(mockConnection);
+
+    this.registry = new StandardServiceRegistryBuilder()
+        .applySetting("hibernate.dialect", SpannerDialect.class.getName())
+        // must NOT set a driver class name so that Hibernate will use java.sql.DriverManager
+        // and discover the only mock driver we have set up.
+        .applySetting("hibernate.connection.url", "unused")
+        .applySetting("hibernate.connection.username", "unused")
+        .applySetting("hibernate.connection.password", "unused")
+        .applySetting("hibernate.hbm2ddl.auto", "update")
+        .build();
+  }
+
+  @Test
+  public void testUpdateStatements_createTables() throws SQLException {
+    // This sets up which pre-existing tables that Hibernate sees.
+    mockConnection.setMetaData(MockJdbcUtils.createMockDatabaseMetaData("Hello"));
+
+    Metadata metadata =
+        new MetadataSources(this.registry)
+            .addAnnotatedClass(Employee.class)
+            .buildMetadata();
+
+    Session session = metadata.buildSessionFactory().openSession();
+    session.beginTransaction();
+    session.close();
+
+    List<String> sqlStrings = mockConnection.getStatementResultSetHandler().getExecutedStatements();
+
+    assertThat(sqlStrings).containsExactly(
+        "START BATCH DDL",
+        "create table Employee (id INT64 not null,name STRING(255),manager_id INT64) "
+            + "PRIMARY KEY (id)",
+        "create table hibernate_sequence (next_val INT64) PRIMARY KEY ()",
+        "INSERT INTO hibernate_sequence (next_val) VALUES(1)",
+        "create index name_index on Employee (name)",
+        "RUN BATCH"
+    );
+  }
+
+  @Test
+  public void testUpdateStatements_alterTables() throws SQLException {
+    // This sets up which pre-existing tables that Hibernate sees.
+    mockConnection.setMetaData(MockJdbcUtils.createMockDatabaseMetaData("Employee"));
```

Comment:
Can you refactor a bit so that the two tests don't have so much duplication?

Reply:
Done.